runtime/v2/shim/shim.go

```diff
@@ -271,11 +272,12 @@ func run(id string, initFunc Init, config Config) error {
 }
 
 // NewShimClient creates a new shim server client
-func NewShimClient(ctx context.Context, svc shimapi.TaskService, signals chan os.Signal) *Client {
+func NewShimClient(ctx context.Context, svc shimapi.TaskService, signals chan os.Signal, shutdown context.CancelFunc) *Client {
```

Comment:
Passing in the cancel here shouldn't be necessary, only the serve function needs to return. This could be done by just creating a cancel in the serve context and passing that cancellable context to reap. The cancellable contexts will inherit cancellation if needed already.

Reply:
updated the method.